model/src/main/kotlin/utils/OrtResultExtensions.kt

```diff
@@ -84,3 +90,35 @@ fun OrtResult.getDetectedLicensesWithCopyrights(
     omitExcluded: Boolean = true
 ): Map<String, Set<String>> =
     LicenseResolver(this, packageConfigurationProvider).getDetectedLicensesWithCopyrights(id, omitExcluded)
+
+/**
+ * Create [PackageConfiguration]s for the projects in this [OrtResult]. [PathExclude]s and [LicenseFindingCuration]s are
+ * taken from the [RepositoryConfiguration]. The [Provenance] is taken from the scan results. Projects without scan
+ * result and scan results without provenance information are ignored.
+ */
+fun OrtResult.getProjectConfigurations(): List<PackageConfiguration> {
+    fun VcsInfo.toMatcher() = VcsMatcher(
+        type = type,
+        url = url,
+        revision = resolvedRevision!!,
+        path = path
+    )
+
+    return analyzer?.result?.projects.orEmpty().flatMap { project ->
+        getScanResultsForId(project.id).mapNotNull { scanResult ->
+            // TODO: Currently the source artifact of the provenance is ignored, because for projects always the VCS is
+            //       scanned. Also, the contents of the source artifact might differ from the file structure of inside
+            //       the VCS and therefore the configuration might not match. If scanning of source artifacts of
+            //       projects is implemented, this needs to be reconsidered.
+            scanResult.provenance.vcsInfo?.toMatcher()?.let { vcsMatcher ->
+                PackageConfiguration(
+                    id = project.id,
+                    sourceArtifactUrl = null,
+                    vcs = vcsMatcher,
+                    pathExcludes = repository.config.excludes.paths,
+                    licenseFindingCurations = repository.config.curations.licenseFindings
```

Comment:
I just noticed here the thing we discussed: the pathExcludes / curations are relative to the analyzer root and not relative to the vcs for nested repositories.

Reply:
Right, as discussed this will be fixed later.